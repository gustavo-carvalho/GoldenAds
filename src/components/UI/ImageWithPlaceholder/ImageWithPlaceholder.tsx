import * as React from 'react'
import { Image, ImageStyle, ImageSourcePropType } from 'react-native'

import R from 'src/res'

export interface IProps {
  source: ImageSourcePropType
  style?: ImageStyle
}

export interface IState {
  error: boolean
}

export class ImageWithPlaceholder extends React.Component<IProps, IState> {
  state = {
    error: false,
  }

  _onLoadError = () => {
    this.setState({ error: true })
  }

  _getImgSource = () => {
    if (this.state.error) {
      return R.images.placeholder
    }

    return this.props.source
  }

  render() {
    return (
      <Image
        style={this.props.style}
        onError={this._onLoadError}
        source={this._getImgSource()}
      />
    )
  }
}
