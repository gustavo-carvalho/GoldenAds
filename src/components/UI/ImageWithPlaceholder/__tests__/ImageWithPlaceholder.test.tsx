import React from 'react'
import renderer from 'react-test-renderer'
import { ImageWithPlaceholder } from '../ImageWithPlaceholder'

describe('ImageWithPlaceholder', () => {
  it('renders correctly', () => {
    const imgUri = 'https://facebook.github.io/react-native/img/header_logo.png'
    const imageWithPlaceholder = renderer
      .create(<ImageWithPlaceholder imgUri={imgUri} />)
      .toJSON()
    expect(imageWithPlaceholder).toMatchSnapshot()
  })
})
