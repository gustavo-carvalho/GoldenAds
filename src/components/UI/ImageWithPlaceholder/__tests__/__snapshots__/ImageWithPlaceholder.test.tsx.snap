// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageWithPlaceholder renders correctly 1`] = `
<Image
  onError={[Function]}
  source={
    Object {
      "uri": "https://facebook.github.io/react-native/img/header_logo.png",
    }
  }
/>
`;
